### DISCUSSION TITLE
How to generate a default config?

### DISCUSSION THREAD

**Discussion by huyz** [kovidgoyal/kitty#5978] [Q&A] [answered]
How can I have Kitty spit out a default config?
When I upgrade Kitty, I'd like to see how the default config has changed so that I can integrate some of the new features into my actual config file. So I need to save a baseline and then, after every upgrade, I would compare the new default config with my baseline and do a three-way merge, e.g. using Meld.

I guess I could move my config out of the way, launch Kitty, invoke the "Settings…" menu item so that a new default `kitty.conf` is generated. But this is even more inconvenient.

Actually, what I really wish is that a default `kitty.conf` be added to the git repo. That way it would be trivial to do a git diff between releases and/or a three-way merge with all 3 files.

**Comment by kovidgoyal** [comment]
https://sw.kovidgoyal.net/kitty/conf/#sample-kitty-conf

On Fri, Feb 03, 2023 at 04:59:58AM -0800, Huy Z wrote:
> How can I have Kitty spit out a default config?
> When I upgrade Kitty, I'd like to see how the default config has changed so that I can integrate some of the new features into my actual config file. So I need to save a baseline and then, after every upgrade, I would compare the new default config with my baseline.
> 
> I guess I could move my config out of the way, launch Kitty, invoke the "Settings…" menu item so that a new default `kitty.conf` is generated. But this is even more inconvenient.
> 
> Actually, what I really wish is that a default `kitty.conf` be added to the git repo. That way it would be trivial to do a git diff between releases.
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/5978
> You are receiving this because you are subscribed to this thread.
> 
> Message ID: ***@***.***>

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Comment by page-down** [comment, marked as answer]
> ... I'd like to see how the default config has changed ...

See the following issue.

https://github.com/kovidgoyal/kitty/issues/628


```python
#!/usr/bin/env kitty +launch
from kitty.config import commented_out_default_config as conf
from kitty.constants import str_version


if __name__ == '__main__':
    file_path = f'kitty.{str_version}.default.conf'
    with open(file_path, 'w', encoding='utf-8') as f:
        f.write(conf())
    print(file_path)
```

```shell
chmod +x generate_default_conf.py
./generate_default_conf.py
```

> Actually, what I really wish is ... do a git diff between releases ...

You can save a commented default configuration file in a Git repository for easy comparison.
If you wish to change configuration options in the commented configuration file, add new lines to configure them instead of uncommenting and changing them directly. This will help with comparisons.

@kovidgoyal
Example configuration file generated in the documentation, where all configuration options are uncommented. If a user uses this file and modifies it, future changes to certain option values will not take effect by default.

For example, a new protocol is appended to `url_prefixes`.

Should we keep the comment in front of the default configuration options to avoid this?
This will make it consistent with what `ctrl+shift+f2` generates.

**Reply by huyz** [reply]
The above script is closest to what I'm looking for, thanks!

**Comment by kovidgoyal** [comment]
On Fri, Feb 03, 2023 at 06:05:28AM -0800, page-down wrote:
> Should we keep the comment in front of the default configuration options to avoid this?
> This will make it consistent with what `ctrl+shift+f2` generates.

Yeah, makes sense.

@OP If you want to see all changes to options simply browse the history
of options/definition.py in github.

**Reply by op** [reply]
@huyz this is your personal bot talking. @kovidgoyal wants your attention! :sparkles:

**Reply by kovidgoyal** [reply]
Oops sorry about that!
